lib/tests/streamlit/streamlit_test.py

```diff
@@ -454,6 +453,48 @@ def test_st_pyplot(self):
         checksum = "iVBORw0KGgoAAAANSUhEUgAAAZAAAAGQCAYAAACAvzb"
         self.assertTrue(el.imgs.imgs[0].data.base64.startswith(checksum))
 
+    def test_st_pyplot_clear_figure(self):
+        """st.pyplot should clear the passed-in figure."""
+        if sys.version_info < (3, 0):
+            # We don't test matplotlib under Python 2, because we're not
+            # able to reliably force the backend to "agg".
+            return
+
+        import matplotlib
+        import matplotlib.pyplot as plt
+
+        if matplotlib.get_backend().lower() != "agg":
+            plt.switch_backend("agg")
+
+        # Assert that plt.clf() is called by st.pyplot() only if
+        # clear_fig is True
+        for clear_fig in [True, False]:
+            plt.hist(np.random.normal(1, 1, size=100), bins=20)
+            with patch.object(plt, "clf", wraps=plt.clf, autospec=True) as plt_clf:
+                st.pyplot(clear_fig=clear_fig)
+
+                if clear_fig:
+                    plt_clf.assert_called_once()
+                else:
+                    plt_clf.assert_not_called()
+
+            # Manually clear for the next test
+            plt.clf()
+
+        # Assert that fig.clf() is called by st.pyplot(fig) only if
+        # clear_fig is True
+        for clear_fig in [True, False]:
```

Comment:
Same test?

Reply:
Yeah, seemed appropriate here, since the tests are tiny and there's some boilerplate. I'd be happy to change if people feel strongly!